Data/Sys/Shaders/default_pre_post_process.glsl

```diff
@@ -195,214 +255,175 @@ float4 SharpBilinearSample(float3 uvw, float gamma)
-
-	float4 color;
-
-	if (needs_resampling)
-	{
-		// Doing linear sampling in "gamma space" on linear texture formats isn't correct.
-		// If the source and target resolutions don't match, the GPU will return a color
-		// that is the average of 4 gamma space colors, but gamma space colors can't be blended together,
-		// gamma neeeds to be de-applied first. This makes a big difference if colors change
-		// drastically between two pixels.
-
-		color = LinearGammaCorrectedSample(game_gamma);
-	}
-	else
-	{
-		// Default GPU HW sampling. Bilinear is identical to Nearest Neighbor if the input and output resolutions match.
-		if (needs_rescaling)
-			color = texture(samp0, v_tex0);
-		else
-			color = texture(samp1, v_tex0);
-
-		// Convert to linear before doing any other of follow up operations.
-		color.rgb = pow(color.rgb, float3(game_gamma));
-	}
-
-	if (OptionEnabled(correct_color_space))
-	{
-		if (game_color_space == 0)
-			color.rgb = color.rgb * from_NTSCM;
-		else if (game_color_space == 1)
-			color.rgb = color.rgb * from_NTSCJ;
-		else if (game_color_space == 2)
-			color.rgb = color.rgb * from_PAL;
-	}
-
-	if (OptionEnabled(hdr_output))
-	{
-		float hdr_paper_white = hdr_paper_white_nits / hdr_sdr_white_nits;
-		color.rgb *= hdr_paper_white;
-	}
-
-	if (OptionEnabled(linear_space_output))
-	{
-		// Nothing to do here
-	}
-	// Correct the SDR gamma for sRGB (PC/Monitor) or ~2.2 (Common TV gamma)
-	else if (OptionEnabled(correct_gamma))
-	{
-		if (OptionEnabled(sdr_display_gamma_sRGB))
-			color.rgb = LinearTosRGBGamma(color.rgb);
-		else
-			color.rgb = pow(color.rgb, float3(1.0 / sdr_display_custom_gamma));
-	}
-	// Restore the original gamma without changes
-	else
-	{
-			color.rgb = pow(color.rgb, float3(1.0 / game_gamma));
-	}
-
-	SetOutput(color);
+    float4 color = float4(0.0, 0.0, 0.0, 1.0);
+
+    if (GetResolution() == GetWindowResolution() || resampling_method == 0)
+        color = sRGBToLinear(texture(samp1, v_tex0));
+    else
+        color = GetResampledColor();
+
+    if (OptionEnabled(correct_color_space)) {
+        if (game_color_space == 0)
+            color.rgb = color.rgb * from_NTSCM;
+        else if (game_color_space == 1)
+            color.rgb = color.rgb * from_NTSCJ;
+        else if (game_color_space == 2)
+            color.rgb = color.rgb * from_PAL;
+    }
+
+    if (OptionEnabled(hdr_output)) {
+        float hdr_paper_white = hdr_paper_white_nits / hdr_sdr_white_nits;
+        hdr_paper_white = pow(hdr_paper_white, 1.0 / 3.0); // Oklab is in a cube root space
+
+        // in order to use the full range of the display at full HDR,
+        // linear brighness needs to go down when the exposure goes up.
+        // This is because scRGB is a -0.5 to 7.5 range.
+        float4 brightness_adjustment = float4((hdr_paper_white - 1) / 6.0, 0.0, 0.0, 0.0);
+        color = OklabToLinearRGB(hdr_paper_white * LinearRGBToOklab(color) - brightness_adjustment);
```

Comment:
This is a very unconventional way of upgrading SDR to HDR and it's subjective. This needs to be a post process effect, akin to the AutoHDR post process shader that Dolphin already has.
This this will also create negative oklab values, due to the subtraction, which could result in very random/invalid colors.

This currently shifts both hue, chroma/saturation and intensity/brightness, thus making it a 100% non conservative SDR to HDR upgrade path. I suggest using Oklch to boost up the chroma and intensity only, without touching the hue, but even then, this is a bit random of a thing to do. I feel like you are just amazed by the boosted perceived saturation and brightness as of now, but that's not really what Dolphin goes for, that should be your own post process that you can use. The artistic intent of games is ruined with this change.

Reply:
Multiplying brightness and chroma in Oklch by an amount is mathematically identical to multiplying all 3 axis in Oklab by that amount. This does not introduce any hue shift and IS a conservative upgrade path. Negative Oklab colors are actually ok, they don't look bad, they just look darker.

I encourage you to actually try this out, the artistic intent of the game is not lost, and it looks like HDR is supposed to. I've shown many people in person and they all agree this is an improvement.

Reply:
you are right, there is no hue shift, just chroma and brightness. 
I tried it. What I said still applies, this a subjective SDR to HDR upgrade, and thus should be in a post process shader file.